torch/distributed/distributed_c10d.py

```diff
@@ -1156,6 +1157,148 @@ def all_gather_multigpu(output_tensor_lists,
         work.wait()
 
 
+def _object_to_tensor(obj):
+    buffer = io.BytesIO()
+    torch.save(obj, buffer)
+    input_tensor = torch.ByteTensor(list(buffer.getvalue()))
+    local_size = torch.LongTensor([input_tensor.numel()])
+    return input_tensor, local_size
+
+
+def _tensor_to_object(tensor, tensor_size):
+    buf = io.BytesIO(tensor.numpy().tobytes()[:tensor_size])
+    buf.seek(0)
+    out = torch.load(buf)
+    return out
+
+
+def all_gather_object(object_list, obj, group=group.WORLD):
+    """
+    Gathers picklable objects from the whole group into a list. Similar to
+    all_gather, but Python objects can be passed in. Note that the object must
+    be pickable in order to be gathered.
+
+    Arguments:
+        object_list (list[Any]): Output list. It should be correctly sized as the
+        size of the group for this collective and will contain the output.
+        object (Any): Pickable Python object to be broadcast from current process.
+        group (ProcessGroup, optional): The process group to work on
+
+    Returns:
+        None. If the calling rank is part of this group, the output of the
```

Comment:
Let's also say that if the rank is not in the group, the `object_list` argument will stay intact.